### DISCUSSION TITLE
Wepack 5 Module Federation Plugin Server Side Remote can use remote address

### DISCUSSION THREAD

**Discussion by Schnaubel** [webpack/webpack#12477] [Q&A]
Hello,

I followed some examples from: https://github.com/module-federation and came across this **server.base.js** ([link to exact exampel](https://github.com/module-federation/module-federation-examples/tree/master/server-side-rendering)) webpack config.

```
const remotePath = path.resolve(
    __dirname,
    "../../../website2/buildServer/container.js"
)
...
new ModuleFederationPlugin({
      name: "website1",
      library: { type: "commonjs2" },
      filename: "container.js",
      remotes: {
        website2: remotePath
...
```

So I wonder if remotePath for the **server side** can also be a real remote address (eg: remote1: 'http://some.cdn.com/server/remote1/remote-entry.js',) like in the clients config. I tried but can't get it woking. If not what would be the recommended way if website1 and website2 are deployed to different servers/pods?

Cheers, Tim

**Comment by sokra** [comment]
The value is basically passed to node.js `require()` and that only supports paths on the local disk.

I would recommend to create (or generate) a package.json for the dist assets like that:

``` json
{
  "dependencies": {
    "@remotes/website2": "http://containers.my.company/website2.tar.gz"
  }
}
```

with

``` js
new ModuleFederationPlugin({
  // ...
  remotes: {
    website2: "@remotes/website2"
  }
  // ...
})
```

and run a `yarn` or `npm` to grab the other containers from their remote locations before running the server. This also saves you from connection issues happening during the server runtime. If the package manager succeed, everything is placed on disk to be consumed.

Note: Alternatively you can use a custom npm registry where you deploy your compiled containers to.

**Reply by Schnaubel** [reply]
Ok thanks. Does this approach mean, that when website2 changes during runtime of website1, the changes would not be reflected in website1?

**Reply by Schnaubel** [reply]
Is it possible to update the container/module reference during runtime so that we don't have to restart the express/server process?

**Reply by alexander-akait** [reply]
Right now no, it requires complex solution, but we are thinking about it

**Reply by Schnaubel** [reply]
Thanks
